src/app/store/reference-data/effects/reference-data.effects.ts

```diff
@@ -22,9 +23,17 @@ export class ReferenceDataEffects {
   fetchReferenceDataByType$ = createEffect(() =>
     this.actions$.pipe(
       ofType(fetchReferenceData),
-      mergeMap(({ resourceType }) =>
-        this.referenceDataService.fetchReferenceData(resourceType).pipe(
-          map(data => fetchReferenceDataSuccess({ resourceType, payload: data.data })),
+      mergeMap(({ resourceType, paginationToken }) =>
+        this.referenceDataService.fetchReferenceData(resourceType, paginationToken).pipe(
+          switchMap(data => {
+            if (isPaginated(data)) {
+              return of(
+                fetchReferenceDataSuccess({ resourceType, payload: data.data as ReferenceDataModelBase[] }),
+                fetchReferenceData({ resourceType, paginationToken: data.paginationToken })
+              );
+            }
+            return of(fetchReferenceDataSuccess({ resourceType, payload: data.data as ReferenceDataModelBase[] }));
+          }),
```

Comment:
Ternary here maybe?
```suggestion
          switchMap(data => isPaginated(data)
            ? of(
              fetchReferenceDataSuccess({ resourceType, payload: data.data as ReferenceDataModelBase[] }),
              fetchReferenceData({ resourceType, paginationToken: data.paginationToken })
            )
            : of(fetchReferenceDataSuccess({ resourceType, payload: data.data as ReferenceDataModelBase[] }))
          ),
```

Reply:
Don't look as readable :(

Reply:
I thought about it, but because its's a sort of long winded expression it feels much less readable.

Reply:
That is fair.